startup.c

```diff
@@ -44,14 +44,18 @@ int main(int argc, char **argv)
 {
 	int c;
 	int standalone = 0;
+	int foreground = 0;
 
 #ifndef HAVE_PROGNAME
 	if (!(progname = argv[0]))
 		progname = POP_SERVER;
 #endif
 
-	while ((c = getopt(argc, argv, "DV")) != -1) {
+	while ((c = getopt(argc, argv, "DFV")) != -1) {
 		switch (c) {
+		case 'F':
+			foreground++;
+			/* [[fallthrough]]; */
```

Comment:
What compilers recognize `/* [[fallthrough]]; */`?

FWIW, in LKRG we define and use this macro:

```c
#if defined(fallthrough)
 #define P_FALL_THROUGH fallthrough
#elif defined(__GNUC__) && __GNUC__ > 6
 #define P_FALL_THROUGH __attribute__ ((fallthrough))
#else
 #define P_FALL_THROUGH /* FALLTHROUGH */
#endif
```

but its `#if defined(fallthrough)` part is specific to the kernel. Of course, I'd rather have just a simple comment understood by compilers than introduce a macro into this project as well... or avoid the fall-through to side-step it.

In yescrypt, including its integration in libxcrypt, it's just `/* FALLTHRU */`, and I don't recall warning reports about that, so apparently it works pretty well.